### PR TITLE
refactor(vars): rewrite/reorganize global vars

### PR DESCRIPTION
**Pull request by macintoshpie**

### CODE REVIEW

File: vars/microservicePipeline.groovy [hunk 1]
```diff
@@ -1,207 +1,40 @@
...
-      stage('GenerateTestData') {
-        // Run the data simulator
-        steps {
-          dir('gen3-qa') {
-            withEnv(['GEN3_NOPROXY=true', "vpc_name=qaplanetv1", "GEN3_HOME=$env.WORKSPACE/cloud-automation", "NAMESPACE=$env.KUBECTL_NAMESPACE", "TEST_DATA_PATH=$env.WORKSPACE/testData/"]) {
-              sh "bash ./jenkins-simulate-data.sh $env.KUBECTL_NAMESPACE"
-            }
-          }
-        }
+      stage('FetchDataClient') {
```
Comment (thanh-nguyen-dang) [inline vars/microservicePipeline.groovy]: duplicated stage?

File: vars/microservicePipeline.groovy [hunk 2]
```diff
@@ -1,207 +1,40 @@
...
+        pipe.kube.reset()
       }
-      stage('VerifyClusterHealth') {
-        steps {
-          withEnv(['GEN3_NOPROXY=true', "vpc_name=qaplanetv1", "GEN3_HOME=$env.WORKSPACE/cloud-automation"]) {
-            sh "bash cloud-automation/gen3/bin/kube-wait4-pods.sh"
-            sh "bash ./gen3-qa/check-pod-health.sh"
-          }
-        }
+      stage('GenerateData') {
```
Comment (thanh-nguyen-dang) [inline vars/microservicePipeline.groovy]: do you have VerifyClusterHealth somewhere?

File: Jenkinsfile [hunk 3]
```diff
@@ -5,8 +5,8 @@
...
 library identifier: "cdis-jenkins-lib@${env.CHANGE_BRANCH}"
 
-testPipeline {
-  // overrides for testing the ModifyManifest stage ...
-  JOB_NAME = 'indexd'
-  GIT_BRANCH = 'master'
+runPipeline {
+  pipeline = 'microservice'
+  serviceTesting = [name: 'fence', branch: 'master']
+  skipQuay = true
```
Comment (frickjack) [inline Jenkinsfile]: Don't `skipQuay` - we want to test that code too ...

File: vars/kubeHelper.groovy [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+@Field def vpcName
+@Field def obtainedLock // indicates if successfully locked a namespace
+
+/**
+* Constructor for kubeHelper
+* Provides access to kubectl and gen3 commands
+*
+* @param config - pipeline config
+*/
+def create(Map config) {
```
Comment (frickjack) [inline vars/kubeHelper.groovy]: What is going on here?  What is `this`?  Why do you call this method a constructor?  Does this all unwind if we start using remote executors?  Do you have a reference to some online docs that encourage using this pattern of decorating `this` with a bunch of global variables?

Comment (macintoshpie) [reply vars/kubeHelper.groovy]: Scripts are compiled into classes by default in groovy, see here http://groovy-lang.org/structure.html#_script_class

Also see here for src vs vars:
- https://www.youtube.com/watch?v=M8U9RyL756U&feature=youtu.be&list=PLvBBnHmZuNQLqgKDFmGnUClw68qsQ9Hq5&t=2310
- http://www.aimtheory.com/jenkins/pipeline/continuous-delivery/2018/10/01/jenkins-pipeline-global-shared-library-best-practices-part-2.html

I experimented with using both versions (that's why I've been working on this for so long), and it seemed to me that this was a pretty good solution. We can chat about this in person if you'd like

**Pull request closed without merging**